histomicstk/segmentation/nuclear/_max_clustering_cython.pyx

```diff
@@ -0,0 +1,145 @@
+import numpy as np
+cimport numpy as np
+cimport cython
+
+@cython.cdivision(True)
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def _max_clustering_cython(double[:, :] im not None, int[:, :] im_fgnd_mask not None, double rad):
+
+    cdef long sx = im.shape[1]
+    cdef long sy = im.shape[0]
+    cdef long r = int(rad + 0.5)
+
+    # find foreground pixels
+    cdef long[:] px, py
+
+    py, px = np.nonzero(im_fgnd_mask)
+
+    cdef long num_pixels = py.shape[0]
+
+    # find local maximum of all requested pixels
+    cdef double[:, ::1] local_max_val = np.zeros([sy, sx], dtype=np.float)
+    cdef long[:, ::1] local_max_ind = np.zeros([sy, sx], dtype=np.int)
+    cdef int[:, ::1] peak_found = np.zeros([sy, sx], dtype=np.int32)
```

Comment:
The type of peak_found seems to be used as 'bool'. Should we use 'bool' type as 'int' from now?